server/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -144,13 +145,24 @@ public IndexModule(
             final IndexSettings indexSettings,
             final AnalysisRegistry analysisRegistry,
             final EngineFactory engineFactory,
-            final Map<String, IndexStorePlugin.DirectoryFactory> directoryFactories) {
+            final Map<String, IndexStorePlugin.DirectoryFactory> directoryFactories,
+            final BooleanSupplier isAllowExpensiveQueries) {
         this.indexSettings = indexSettings;
         this.analysisRegistry = analysisRegistry;
         this.engineFactory = Objects.requireNonNull(engineFactory);
         this.searchOperationListeners.add(new SearchSlowLog(indexSettings));
         this.indexOperationListeners.add(new IndexingSlowLog(indexSettings));
         this.directoryFactories = Collections.unmodifiableMap(directoryFactories);
+        this.isAllowExpensiveQueries = isAllowExpensiveQueries;
+    }
+
+    // For testing
+    IndexModule(
+            final IndexSettings indexSettings,
+            final AnalysisRegistry analysisRegistry,
+            final EngineFactory engineFactory,
+            final Map<String, IndexStorePlugin.DirectoryFactory> directoryFactories) {
+        this(indexSettings, analysisRegistry, engineFactory, directoryFactories, () -> true);
     }
```

Comment:
can't we have tests call the other constructor instead and pass `()->true`?

Reply:
Yep, that's easy to do.